lib/internal/readme.md

```diff
@@ -0,0 +1,4 @@
+# Internal Modules
+
+These modules are intended for internal use in node.js core only, and are not accessible with `require()` from user modules.
+Your warranty on API breakage is **void** if you directly use **anything** in `lib/internal` outside node core.
```

Comment:
Maybe say something like "These are subject to change at any time. Reliance on these modules outside of core is **not supported**."